src/Nethermind/Nethermind.Consensus.AuRa/Transactions/PermissionTxComparerBase.cs

```diff
@@ -0,0 +1,54 @@
+﻿//  Copyright (c) 2018 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using System.Collections.Generic;
+using Nethermind.Core;
+using Nethermind.Int256;
+
+namespace Nethermind.Consensus.AuRa.Transactions
+{
+    public abstract class PermissionTxComparerBase : IComparer<Transaction>
```

Comment:
I think I like it

Reply:
great it got pushed to AuRa

Reply:
made it a bit better

Reply:
I liked the cimplicity. Now it feels like it suddenly does it a bit more.

Reply:
yes but managed to get rid of factory